### PR TITLE
[fastlane] Crashlytics upload now supports zip files and nested dSYMs

### PR DESCRIPTION
**Pull request by KrauseFx**
Oh yeah for danger

### CODE REVIEW

File: fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb [hunk 1]
```diff
@@ -2,9 +2,64 @@ module Fastlane
...
-        UI.user_error!("Please provide a path to the binary using binary_path:") unless params[:binary_path]
+        require 'tmpdir'
+
+        find_binary_path(params)
+        find_api_token(params)
+
+        dsym_paths = []
+        dsym_paths << params[:dsym_path]
+        dsym_paths += Actions.lane_context[SharedValues::DSYM_PATHS] if Actions.lane_context[SharedValues::DSYM_PATHS]
+        UI.error("Couldn't find any dSYMs, please pass them using the dsym_path option") if dsym_paths.count == 0
```
Comment (asfalcone) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: Should this be `UI.user_error!`? If there are no dsym_paths then this action can't proceed from here.

Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: The reason for this: You might want to check for dSYMs and upload them if available (e.g. running on a cron job). So this shouldn't fail, however I see that this should jump out of the action, to not show a success message. 
Does that make sense?

File: fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb [hunk 2]
```diff
@@ -2,9 +2,64 @@ module Fastlane
...
+          end
+        else
+          UI.error "Don't know how to handle '#{current_path}'"
+        end
+      end
+
+      def self.upload_dsym(params, path)
+        UI.message("Uploading '#{path}'...")
+        command = []
+        command << File.expand_path(params[:binary_path])
```
Comment (asfalcone) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: Has this already been expanded at https://github.com/fastlane/fastlane/pull/4017/files#diff-2eec78d70c41940712d3172fc80d1898R78?

File: fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb (hunk 1 above)
Comment (asfalcone) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: That use case is interesting and certainly makes life easier for the user who doesn't have to check if there are dsyms before calling upload. Let's return after this if no dsyms.

File: fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb (hunk 2 above)
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: Good point, I'll update the PR

**Comment by asfalcone** [comment]
LGTM after those 2 comments! :rocket:

File: fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb (hunk 1 above)
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: Yeah! :+1:

**Comment by KrauseFx** [comment]
Alright, updated the PR, let's wait for CI :+1:

**Comment by KrauseFx** [comment]
And here we go :rocket:

**Pull request merged**